azext_iot/deviceupdate/commands_instance.py

```diff
@@ -0,0 +1,120 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.log import get_logger
+from azext_iot.common.utility import handle_service_exception
+from azext_iot.deviceupdate.providers.base import (
+    DeviceUpdateMgmtModels,
+    DeviceUpdateInstanceManager,
+    parse_account_rg,
+)
+
+logger = get_logger(__name__)
+
+# Instances
+
+
+def create_instance(
+    cmd,
+    name,
+    instance_name,
+    iothub_resource_ids,
+    diagnostics=None,
+    storage_resource_id=None,
+    location=None,
+    tags=None,
+    resource_group_name=None,
+):
+    instance_manager = DeviceUpdateInstanceManager(cmd=cmd)
+    target_container = instance_manager.find_account(target_name=name, target_rg=resource_group_name)
+    if not location:
+        location = target_container.account.location
+
+    instance = DeviceUpdateMgmtModels.Instance(
+        location=location,
+        tags=tags,
+        iot_hubs=instance_manager.assemble_iothub_resources(iothub_resource_ids),
+        enable_diagnostics=diagnostics,
+    )
+    if storage_resource_id:
+        instance.diagnostic_storage_properties = instance_manager.assemble_diagnostic_storage(storage_resource_id)
+    try:
+        return instance_manager.mgmt_client.instances.begin_create(
+            resource_group_name=target_container.resource_group,
+            account_name=target_container.account.name,
+            instance_name=instance_name,
+            instance=instance,
+        )
+    except Exception as e:
+        handle_service_exception(e)
+
+
+def update_instance(cmd, parameters: DeviceUpdateMgmtModels.Instance):
+    instance_manager = DeviceUpdateInstanceManager(cmd=cmd)
+
+    try:
+        storage_properties = parameters.diagnostic_storage_properties
+        if storage_properties:
+            if (
```

Comment:
you can just combine the two if's

Reply:
Keeping `if storage_properties:` at the separate scope allows for additional independent conditional checks of storage_properties members, but it is true there are no additional checks happening so I combined them.